check for PhysicalFileProvider

stefannikolei opened the pull request:
* check that it is not a subclass of PhysicalFileProvider
fix #47455

stefannikolei committed:
check for PhysicalFileProvider

* check that it is not a subclass of PhysicalFileProvider

maryamariyan commented:
@stefannikolei thanks for the PR could you also please add a test for this?

maryamariyan reviewed (changes requested):
The PR change seems fine, @stefannikolei could you please also add a tests for this fix?

eerhardt commented on src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationProvider.cs:
```diff
@@ -76,9 +76,10 @@ private void Load(bool reload)
                 {
-                    if (fileInfo.PhysicalPath != null)
+                    if (fileInfo.PhysicalPath != null
+                        && provider is PhysicalFileProvider && !provider.GetType().IsSubclassOf(typeof(PhysicalFileProvider)))
```
Couldn't this just be

```suggestion
                        && provider?.GetType() == typeof(PhysicalFileProvider))
```

Also - would be great to add a comment here why we are doing this.

davidfowl replied:
Can we do this check once in the constructor

safern commented:
@stefannikolei thanks for your PR, do you plan on addressing the feedback and adding a test case for this?

stefannikolei commented:
> @stefannikolei thanks for your PR, do you plan on addressing the feedback and adding a test case for this?

Sorry missed the feedback. Gonna look at it.
Was waiting for @mcm-ham to hopefully deliver some repro code to make the test easier.

maryamariyan commented:
Closing this PR. @stefannikolei this is a good change, you could create a new PR if you plan on addressing the feedback
